libs/sdk-core/src/lnurl/pay.rs

```diff
@@ -141,12 +140,21 @@ pub(crate) mod model {
     /// field with the reason.
     #[derive(Debug, Serialize, Deserialize)]
     pub enum LnUrlPayResult {
-        EndpointSuccess {
-            data: Option<SuccessActionProcessed>,
-        },
-        EndpointError {
-            data: LnUrlErrorData,
-        },
+        EndpointSuccess { data: LnUrlSuccessData },
+        EndpointError { data: LnUrlErrorData },
+        PayError { data: LnUrlPayErrorData },
```

Comment:
@dleutenegger Useful new error type! Can you please add a line about it in the method's rustdoc?

Reply:
This makes me think that perhaps the right way here is to return the errors as rust errors as right now we are mixing both success and error in the success case (We don't use rust error here).
I do understand the change as the mixture has been existed before this change so I am OK to leave it as is but perhaps after we merge the error code PR @dangeross we can think of changing here to use rust errors if that make sense of course?

Reply:
I don't think the error code PR would give a clean way to return the payment hash (and the reason) of the failed payment, which was the intent of the original issue.